### PR TITLE
support k8s 1.30 and 1.31

### PR DESCRIPTION
**Pull request by YZ775**

### CODE REVIEW

**Review by chez-shanpu — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -26,8 +26,8 @@ Blog article: [Introducing MOCO, a modern MySQL operator on Kubernetes](https://
 
 ## Supported software
 
-- MySQL: 8.0.28, 8.0.36, 8.0.37, 8.4.0
-- Kubernetes: 1.27, 1.28, 1.29
+- MySQL: 8.0.28, 8.0.36, 8.0.37, 8.4.0, 8.4.2
+- Kubernetes: 1.29, 1.30,1.31
```
[the remark this answers is not in the document]
Comment (YZ775) [reply README.md]: Thanks! I fixed these.

**Review by chez-shanpu — APPROVED** [review]
LGTM

File: .github/workflows/ci-e2e.yaml [hunk 2]
```diff
@@ -20,12 +20,12 @@ jobs:
     name: Integration tests with MySQL
     strategy:
       matrix:
-        mysql-version: ["8.0.28", "8.0.36", "8.0.37", "8.4.0"]
+        mysql-version: ["8.0.28", "8.0.36", "8.0.37", "8.4.0", "8.4.2"]
```
Comment (ymmt2005) [inline .github/workflows/ci-e2e.yaml:23]: Isn't our policy support only the latest LTS release?
I mean, we do not need to keep `8.4.0` here.
```suggestion
        mysql-version: ["8.0.28", "8.0.36", "8.0.37", "8.4.2"]
```

File: .github/workflows/ci-e2e.yaml [hunk 3]
```diff
@@ -57,20 +57,21 @@ jobs:
...
+          name: k8s-version-e2e
         if: steps.changed-files.outputs.any_changed == 'true'
 
   e2e-mysql:
     name: Supported MySQL versions End-to-End Tests
     strategy:
       matrix:
-        mysql-version: ["8.0.28", "8.0.36", "8.0.37", "8.4.0"]
-        k8s-version: ["1.29.4"]
+        mysql-version: ["8.0.28", "8.0.36", "8.0.37", "8.4.0", "8.4.2"]
```
Comment (ymmt2005) [inline .github/workflows/ci-e2e.yaml:67]: ditto

File: DEVELOP.md [hunk 4]
```diff
@@ -84,23 +84,23 @@ MySQL versions appear twice:
     name: Integration tests with MySQL
     strategy:
       matrix:
-        mysql-version: ["8.0.28", "8.0.36", "8.0.37", "8.4.0"]
+        mysql-version: ["8.0.28", "8.0.36", "8.0.37", "8.4.0", "8.4.2"]
```
Comment (ymmt2005) [inline DEVELOP.md:87]: ditto

File: DEVELOP.md [hunk 5]
```diff
@@ -84,23 +84,23 @@ MySQL versions appear twice:
...
+        k8s-version: ["1.29.4", "1.30.4", "1.31.0"]
 ...
   # Matrix tests for different MySQL versions on the latest supported Kubernetes version.
   e2e-mysql:
     name: Supported MySQL versions End-to-End Tests
     strategy:
       matrix:
-        mysql-version: ["8.0.28", "8.0.36", "8.0.37", "8.4.0"]
-        k8s-version: ["1.29.4"]
+        mysql-version: ["8.0.28", "8.0.36", "8.0.37", "8.4.0", "8.4.2"]
```
Comment (ymmt2005) [inline DEVELOP.md:102]: ditto

File: README.md [hunk 6]
```diff
@@ -26,8 +26,8 @@ Blog article: [Introducing MOCO, a modern MySQL operator on Kubernetes](https://
 
 ## Supported software
 
-- MySQL: 8.0.28, 8.0.36, 8.0.37, 8.4.0
-- Kubernetes: 1.27, 1.28, 1.29
+- MySQL: 8.0.28, 8.0.36, 8.0.37, 8.4.0, 8.4.2
```
Comment (ymmt2005) [inline README.md:29]:
```suggestion
- MySQL: 8.0.28, 8.0.36, 8.0.37, 8.4.2
```

File: .github/workflows/ci-e2e.yaml (hunk 2 above)
Comment (shunki-fujita) [reply .github/workflows/ci-e2e.yaml:23]: Also write `8.0.39`

**Review by chez-shanpu — APPROVED** [review]
LGTM

**Review by ymmt2005 — DISMISSED** [review]
LGTM

**Review by shunki-fujita — DISMISSED** [review]

**Comment by YZ775** [comment]
There are no changes  that will cause StatafulSet to restart since latest release. So I reverted update of moco-agent.

**Review by chez-shanpu — APPROVED** [review]
LGTM

**Pull request merged**